zest/releaser/utils.py

```diff
@@ -31,6 +49,42 @@ def loglevel():
     return logging.INFO
 
 
+def read_text_file(filename, encoding=None):
+    # Unless specified manually, We have no way of knowing what text
+    # encoding this file may be in.
+    # The 'open' method uses the default system encoding to read text
+    # files in Python 3 or falls back to utf-8.
```

Comment:
Indeed. IIRC parts of chardet did so on second thoughts, I should really
use those where available.
On 2 Jun 2015 7:54 am, "Marius Gedminas" <notifications@github.com> wrote:

> In zest/releaser/utils.py
> <https://github.com/zestsoftware/zest.releaser/pull/97#discussion_r31496561>
> :
>
> > @@ -31,6 +49,42 @@ def loglevel():
> >      return logging.INFO
> >
> >
> > +def read_text_file(filename, encoding=None):
> > +    # Unless specified manually, We have no way of knowing what text
> > +    # encoding this file may be in.
> > +    # The 'open' method uses the default system encoding to read text
> > +    # files in Python 3 or falls back to utf-8.
>
> It's a shame that tokenize.open
> <https://docs.python.org/3.4/library/tokenize.html#tokenize.open> and
> tokenize.detect_encoding
> <https://docs.python.org/3.4/library/tokenize.html#tokenize.detect_encoding>
> never made it into Python 2.7.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zestsoftware/zest.releaser/pull/97/files#r31496561>.
>